### ISSUE TITLE
What-changed CompareDocuments panics

### ISSUE THREAD

**Issue by hugoboos** [pb33f/libopenapi#61]
When comparing the two documents with `what-changed`, the call to `CompareDocuments` panics. It happens when `TotalChanges` is called.

The `ExampleChanges` in `MediaTypeChanges` contains key `test` but the value is `nil`.

```go
func (m *MediaTypeChanges) TotalChanges() int {
    c := m.PropertyChanges.TotalChanges()
    for k := range m.ExampleChanges {
        c += m.ExampleChanges[k].TotalChanges()   <-- panics
    }
    ...
}
```

```yaml
openapi: "3.0.0"
paths:
  /:
    post:
      requestBody:
        content:
          application/json:
            examples:
              test:
                value:
                  date: 2020-01-01  # not quoted
```

```yaml
openapi: "3.0.0"
paths:
  /:
    post:
      requestBody:
        content:
          application/json:
            examples:
              test:
                value:
                  date: "2020-01-01"  # quoted
```

**Comment by daveshanley** [comment]
I will investigate, thanks for finding this.

**Comment by hugoboos** [comment]
Found a possible cause on https://github.com/pb33f/libopenapi/blob/main/what-changed/model/example.go#L73.

The `Tag` of the `Value` is not taken in to account. `!!timestamp` vs `!!str`.

I will create a PR.

**Comment by hugoboos** [comment]
This is not fixed with my fix in #62.

The test case didn't include a map as a value:

```yaml
value:
  map:
    date: 2022-12-29`
```

The `changes := CompareExamples(&lDoc, &rDoc)` returns `nil`. In places where it is used, there is no nil check: https://github.com/pb33f/libopenapi/blob/main/what-changed/model/media_type.go#L26

It is assigned by `CheckMapForChangesWithComp` on https://github.com/pb33f/libopenapi/blob/main/what-changed/model/comparison_functions.go#L253.

So there are a few places where it can break and this might not only be related to examples.

**Comment by daveshanley** [comment]
I will check for other instances of this issue.

**Comment by daveshanley** [comment]
This is an isolated incident; after reviewing the other models, examples are the only place where this type of comparison is performed. The glitch was a `nil` reference being added to the resulting map, as you detected, the code now performs a check before adding the results, and examples that use complex shapes are now hashed for comparison. The fix is in #62
